kong/tools/utils.lua

```diff
@@ -9,45 +9,68 @@
 -- @module kong.tools.utils
 
 local url = require "socket.url"
+local ffi = require "ffi"
 local uuid = require "resty.jit-uuid"
 local pl_stringx = require "pl.stringx"
-local ffi = require "ffi"
 
-local fmt = string.format
-local type = type
-local pairs = pairs
-local ipairs = ipairs
-local re_find = ngx.re.find
-local tostring = tostring
-local table_sort = table.sort
-local table_concat = table.concat
-local table_insert = table.insert
-local string_find = string.find
+local ffi_new    = ffi.new
+local ffi_str    = ffi.string
+local ffi_typeof = ffi.typeof
+local C          = ffi.C
+local fmt        = string.format
+local type       = type
+local pairs      = pairs
+local ipairs     = ipairs
+local re_find    = ngx.re.find
+local tostring   = tostring
+local sort       = table.sort
+local concat     = table.concat
+local insert     = table.insert
+local find       = string.find
+local gsub       = string.gsub
 
 ffi.cdef[[
+typedef unsigned char u_char;
+u_char * ngx_hex_dump(u_char *dst, const u_char *src, size_t len);
+int RAND_bytes(u_char *buf, int num);
+
 int gethostname(char *name, size_t len);
 ]]
 
+local t = ffi_typeof "uint8_t[?]"
```

Comment:
saw the link, was more thinking like; this code was copied/modified from an original by <xyz>, see <link>. To provide proper credits. Possibly mentioning the original license.